src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureNormalizers.java

```diff
@@ -0,0 +1,197 @@
+            // this seems really intended for switching on the key (here featureName) and making
+            // a decision, when in reality, we want to look a layer deeper and switch on that
+            ObjectParser<FeatureNormConsumer, Void> parser = new ObjectParser<>("feature_normalizers",
+                                                                                  FeatureNormConsumer::new);
+
+            BiConsumer<FeatureNormConsumer, StandardFeatureNormDefinition> setStd = (s, v) -> {
+                v.setFeatureName(featureName);
+                s.setFtrNormDefn(v);
+            };
+
+            BiConsumer<FeatureNormConsumer, MinMaxFeatureNormDefinition> setMinMax = (s, v) -> {
+                v.setFeatureName(featureName);
+                s.setFtrNormDefn(v);
+            };
+
+            parser.declareObject(setStd, StandardFeatureNormDefinition.PARSER, STANDARD);
+            parser.declareObject(setMinMax, MinMaxFeatureNormDefinition.PARSER, MIN_MAX);
+
+            FeatureNormConsumer parsedNorm  = parser.parse(p, c);
+            return parsedNorm.ftrNormDefn;
+
+        };
+
+    }
+
+    // A temp holder for parsing out of xcontent
+    private static class FeatureNormConsumer {
+        String featureName;
+        FeatureNormDefinition ftrNormDefn;
+
+        FeatureNormConsumer() {
+        }
+
+        public FeatureNormDefinition getFtrNormDefn() {
+            return this.ftrNormDefn;
+        }
+
+        public void setFtrNormDefn(FeatureNormDefinition ftrNormDefn) {
+            this.ftrNormDefn = ftrNormDefn;
+        }
+    }
+
+
+    private Map<String, FeatureNormDefinition> featureNormalizers;
+
+    public StoredFeatureNormalizers() {
+        this.featureNormalizers = new HashMap<>();
+    }
+
+    public StoredFeatureNormalizers(StreamInput input) throws IOException {
+        this.featureNormalizers = new HashMap<>();
+        int numFeatureNorms = input.readInt();
+        for (int i = numFeatureNorms; i > 0; i--) {
+            FeatureNormDefinition norm = this.createFromStreamInput(input);
+            this.featureNormalizers.put(norm.featureName(), norm);
+        }
+    }
+
+    public Normalizer getNormalizer(String featureName) {
+        return this.featureNormalizers.get(featureName).createFeatureNorm();
+    }
+
+    public static final String TYPE = "feature_normalizers";
+
+
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject(); // begin feature norms
+        for (Map.Entry<String, FeatureNormDefinition> ftrNormDefEntry: featureNormalizers.entrySet()) {
+            builder.field(ftrNormDefEntry.getKey());
+            ftrNormDefEntry.getValue().toXContent(builder, params);
+        }
+        builder.endObject(); // feature norms
+        return builder;
+    }
+
+    public void addFeatureNormalizers(List<FeatureNormDefinition> ftrNormDefs) {
```

Comment:
I think it'd make more sense to have a constructor that accepts a List<FeatureNormDefinition> ftrNormDefs instead of a add method
It seems that whenever we call this method we are actually constructing a new StoredFeatureNormalizers